### ISSUE TITLE
License contains non-free restrictions

### ISSUE THREAD

**Issue by limburgher** [upx/upx#608]
## What's the problem (or question)?
The LICENSE file contains two statements possibly in conflict with distribution and use of UPX in Fedora.
https://github.com/upx/upx/blob/devel4/LICENSE#L99
additionally:
"You can use a modified UPX version or modified UPX stub only for
programs that are compatible with the GNU General Public License."

## What should have happened?
Restrictions of this nature make the software non-compliant with Free Software licensing. Other Linux distributions may also have issues with these stipulations.

## Do you have an idea for a solution?
Removing these stipulations would allow Fedora to continue to ship UPX. If not, we'll have to remove it.

## How can we reproduce the issue?
N/A

## Please tell us details about your environment.
N/A

**Comment by ismail** [comment]
@markus-oberhumer This seems to be a valid issue, could you please add a comment at least?

**Comment by markus-oberhumer** [comment]
UPX is licensed under the GPLv2+, and that's exactly what the annotation says - so I don't get your point about "non-free".

**Comment by ismail** [comment]
@markus-oberhumer Appreciate you taking the time to reply to this. I agree some context is missing here. Original Fedora bug report is at https://bugzilla.redhat.com/show_bug.cgi?id=2133058, comment 1 says:

```
Taken as a whole, the licensing terms of UPX are non-FOSS. In particular, note the "annotation" which says:

"We grant you special permission to freely use and distribute all UPX
compressed programs. But any modification of the UPX stub (such as,
but not limited to, removing our copyright string or making your
program non-decompressible) will immediately revoke your right to
use and distribute a UPX compressed program."
https://github.com/upx/upx/blob/devel4/LICENSE#L99
```

IANAL, but I think that extra requirement about the stub makes it non-FOSS.

**Comment by markus-oberhumer** [comment]
Well, that annotation obviously refers to the "SPECIAL EXCEPTION FOR COMPRESSED EXECUTABLES".

**Comment by ismail** [comment]
@markus-oberhumer Thank you for your answer, I forwarded it to Fedora people so hopefully they reinstate UPX, if not, it's their loss unfortunately.

**Comment by markus-oberhumer** [comment]
I've also updated the docs to make it more clear the UPX is GPLv2+ (COPYING) and that the Special Exceptions (LICENSE) are "at your option", so that should hopefully fix any possible conflicts or misinterpretations.

**Comment by markus-oberhumer** [comment]
@limburgher Your original issue report was not very helpful (and I didn't get what you meant) - why didn't you link to the Fedora bug?

**Comment by markus-oberhumer** [comment]
I think most people always have understood the idea behind our LICENSE, but I can see that it might be open to other unintended legal interpretations.

That's why I have updated the documentation files ([README](https://github.com/upx/upx/blob/master/README), doc/upx.pod) and home page to explicitly make the LICENSE optional:

```
UPX is distributed with full source code under the terms of the
GNU General Public License v2+; either under the pure GPLv2+ (see
the file COPYING), or (at your option) under the GPLv+2 with special
exceptions and restrictions granting the free usage for all binaries
including commercial programs (see the file LICENSE).
```

Hopefully these changes do fix any possible conflicts or misinterpretations.

Just for reference, please note that these 2 files are the only ones that actually mention `LICENSE`  - all other files (including all source code) only ever references the GPLv2+ `COPYING`.

**Comment by richardfontana** [comment]
@markus-oberhumer Yes, thank you, I think that makes it sufficiently clear to be okay for Fedora. cc: @limburgher @ismail

**Comment by markus-oberhumer** [comment]
Great, thanks to everyone helping to resolve this issue!
